### PR TITLE
Close CPPL-67

### PR DESCRIPTION
**Pull request by sorbeee**

### CODE REVIEW

File: src/Server/Handlers/Product/UpdateProductHandler.h [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+#include <string>
+#include <vector>
+
+#include "../AuthorizedHandler.h"
+#include "../libdal/DTOs/Product.h"
+#include "../libdal/Facade/IDbFacade.h"
+
+
+class UpdateProductHandler : public AuthorizedHandler
```
Comment (Denys-Bykov-0) [inline src/Server/Handlers/Product/UpdateProductHandler.h:11]: Wrong outdated handler design, Why do you think we are trying to move parsers to a separate class? Anyway this class development is currently blocked.

File: src/Server/Handlers/Product/UpdateProductHandler.cpp [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+Product UpdateProductHandler::JSONParser::Parse(const QJsonObject &payload)
+{
+	return ParseProduct(payload);
+}
+
+Net::Response UpdateProductHandler::AuthHandle(const Net::Request &request)
+{
+	if(request.method == Net::HTTP_METHOD_PUT)
```
Comment (AndriyFedorenko) [inline src/Server/Handlers/Product/UpdateProductHandler.cpp:23]:
```
if(request.method == Net::HTTP_METHOD_PUT)
 return FormErrorResponse(
		NetError::Status::HTTP_BAD_REQUEST,
		"Unsupported method"
		);
```

File: src/Server/Handlers/Product/UpdateProductHandler.cpp [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	{
+		auto json = request.json_payload.object();
+		try
+		{
+			if(m_facade->UpdateProduct(m_parser.Parse(json)))
+			{
+				return FormEmptyResponse();
```
Comment (AndriyFedorenko) [inline src/Server/Handlers/Product/UpdateProductHandler.cpp:30]: is it valid logic to return empty response ?

Comment (AndriyFedorenko) [reply src/Server/Handlers/Product/UpdateProductHandler.cpp:30]: Don't you need return any data?

**Pull request closed without merging**